File: java/linkedlists/DNode.java
public class DNode<E extends Comparable> {
    private E element;
    private DNode<E> next,prev;

    public DNode(E s, DNode<E> prev, DNode<E> next) {
        element = s;
        this.prev = prev;
        this.next = next;
    }

    public DNode(E s) {
        element = s;
        next=prev=null;
    }

    public E getElement() { return element; }

    public DNode<E> getPrev() { return prev; }

    public DNode<E> getNext() { return next; }

    public void setElement(E s) { element = s; }

    public void setPrev(DNode<E> prev) { this.prev = prev; }

    public void setNext(DNode<E> next) { this.next = next; }

    public String toString() { return element.toString(); }

    public int compareTo(E s) { return element.compareTo(s); }

}

File: python/data-structures/dnode.py
class DNode:
    def __init__(self,elem=None,prev=None,next=None):
        self.elem = elem
        self.prev = prev
        self.next = next

    def get_elem(self):
        return self.elem
    
    def get_next(self):
        return self.next
    
    def get_prev(self):
        return self.prev
    
    def set_elem(self,elem):
        self.elem = elem
    
    def set_next(self,next):
        self.next = next
    
    def set_prev(self,prev):
        self.prev = prev
    
    def __str__(self):
        return self.elem
